app-frontend/src/app/services/scenes/planetRepository.service.js

```diff
@@ -0,0 +1,291 @@
+                        }
+                    });
+                };
+
+                return () => {
+                    if (!scenePages.length && (nextLink === null || nextLink)) {
+                        return search();
+                    }
+                    return this.$q((resolve, reject) => {
+                        if (!scenePages.length) {
+                            reject('No more scenes to fetch');
+                        }
+                        let scenes = scenePages.shift();
+                        let hasNext = scenePages.length || !!nextLink;
+                        resolve({
+                            scenes,
+                            hasNext,
+                            count: nextLink ? `>${fetchedScenes}` : fetchedScenes
+                        });
+                    });
+                };
+            };
+
+            return fetchForBbox;
+        }
+
+        getThumbnail(scene) {
+            return this.$q((resolve, reject) => {
+                this.planetLabsService.getThumbnail(
+                    this.planetToken, scene.thumbnails[0].url
+                ).then((thumbnail) => {
+                    resolve('data:image/png;base64,' + thumbnail);
+                }, (err) => {
+                    reject(err);
+                });
+            });
+        }
+
+        getPreview(scene) {
+            return this.getThumbnail(scene);
+        }
+
+        getDatasource(scene) {
+            return this.$q((resolve) => {
+                let source = _.first(this.sources.filter((s) => s.name === scene.datasource));
+                if (source) {
+                    resolve(source.label);
+                } else {
+                    resolve(scene.datasource);
+                }
+            });
+        }
+
+        addToProject(projectId, scenes) {
+            const planetSceneId = scene => `${scene.datasource}:${scene.id}`;
+            const getDatasourceId =
+                  datasource => _.first(_.filter(this.sources, s => s.id === datasource)).uuid;
+            return this.$q((resolve, reject) => {
+                this.authService.getCurrentUser().then(user => {
+                    // create separate upload for each datasource
+                    let sceneGroups = _.groupBy(scenes, (scene) => scene.datasource);
+                    let datasourceIds = _.map(sceneGroups, (datasourceScenes, datasource) => {
+                        let sceneIds = datasourceScenes.map(planetSceneId);
+                        return {datasource, sceneIds};
+                    });
+                    let uploadPromises = datasourceIds.map(({datasource, sceneIds}) => {
+                        let dsId = getDatasourceId(datasource);
+                        let uploadObject = {
+                            files: sceneIds,
+                            fileType: 'GEOTIFF',
+                            datasource: dsId,
+                            uploadStatus: 'UPLOADING',
```

Comment:
I think this needs to be set to `UPLOADED` else upload processing isn't actually kicked off

Reply:
:man_facepalming: